tools/cosmovisor/scanner.go

```diff
@@ -142,27 +148,56 @@ func (fw *fileWatcher) CheckUpdate(currentUpgrade upgradetypes.Plan) bool {
 	return false
 }
 
-func parseUpgradeInfoFile(filename string) (upgradetypes.Plan, error) {
-	var ui upgradetypes.Plan
+// checkHeight checks if the current block height
+func (fw *fileWatcher) checkHeight() (int64, error) {
+	// TODO(@julienrbrt) use `if !testing.Testing()` from Go 1.22
+	// The tests from `process_test.go`, which run only on linux, are failing when using `autod` that is a bash script.
+	// In production, the binary will always be an application with a status command, but in tests it isn't not.
+	if strings.HasSuffix(os.Args[0], ".test") {
+		return 0, nil
+	}
 
-	f, err := os.Open(filename)
+	result, err := exec.Command(fw.currentBin, "status").Output() //nolint:gosec // we want to execute the status command
+	if err != nil {
+		return 0, err
+	}
+
+	type response struct {
+		SyncInfo struct {
+			LatestBlockHeight string `json:"latest_block_height"`
+		} `json:"SyncInfo"`
+	}
+
+	var resp response
+	if err := json.Unmarshal(result, &resp); err != nil {
+		return 0, err
+	}
+
+	if resp.SyncInfo.LatestBlockHeight == "" {
+		return 0, errors.New("latest block height is empty")
+	}
+
+	return strconv.ParseInt(resp.SyncInfo.LatestBlockHeight, 10, 64)
+}
+
+func parseUpgradeInfoFile(filename string) (upgradetypes.Plan, error) {
+	f, err := os.ReadFile(filename)
 	if err != nil {
 		return upgradetypes.Plan{}, err
 	}
-	defer f.Close()
 
-	d := json.NewDecoder(f)
-	if err := d.Decode(&ui); err != nil {
+	var upgradePlan upgradetypes.Plan
+	if err := json.Unmarshal(f, &upgradePlan); err != nil {
 		return upgradetypes.Plan{}, err
 	}
 
 	// required values must be set
-	if ui.Height <= 0 || ui.Name == "" {
-		return upgradetypes.Plan{}, fmt.Errorf("invalid upgrade-info.json content; name and height must be not empty; got: %v", ui)
+	if err := upgradePlan.ValidateBasic(); err != nil {
+		return upgradetypes.Plan{}, fmt.Errorf("invalid upgrade-info.json content: %w, got: %v", err, upgradePlan)
 	}
 
 	// normalize name to prevent operator error in upgrade name case sensitivity errors.
-	ui.Name = strings.ToLower(ui.Name)
+	upgradePlan.Name = strings.ToLower(upgradePlan.Name)
```

Comment:
We've seen a number of validators fail upgrades due to "not lowercasing" their directory names (by following the plan Name exactly with upper and lower case characters)

Does ToLower() actually introduce a case sensitivity problem?

Reply:
For what it is worth, people had issue with the reverse problem. I do not think the SDK should enforce that at protocol level.
Now that there is a new add-upgrade command in the next version of cosmovisor, we will advise validator to use this command directly when adding a binary (when auto download is disabled). That command create a directory with the correct directory name directly.

Reply:
Since there is no case enforcement at the sdk level, is there a good way to configure cosmovisor to handle plan names: "nextVersion" and "NextVersion" (since both will look in the same ToLower() "nextversion" directory)?

Reply:
I think it should just be a best practice not to have different upgrades with the same name but a different casing.

Reply:
Agreed, I believe that should be avoided!

Reply:
Haha, yeah why not, feel free to add a cosmovisor env for that.